src/core/annotation.js

```diff
@@ -1367,6 +1375,71 @@ class WidgetAnnotation extends Annotation {
     }
     return localResources || Dict.empty;
   }
+
+  _collectJS(entry, xref, list, parents) {
+    if (!entry) {
+      return;
+    }
+
+    let parent = null;
+    if (isRef(entry)) {
+      if (parents.has(entry)) {
+        // If we've already met entry then we've a cycle
+        return;
+      }
+      parent = entry;
+      parents.add(parent);
+      entry = xref.fetch(entry, true);
```

Comment:
It's extremely rare that you need to manually set the `suppressEncryption` parameter when calling `XRef.fetch`, and in general it *may* not even be correct to use it unless you're fetching top-level (in the PDF document tree) data.

Is there an actual test-case that necessitated this?